Assets/_Game/Scripts/UI/ActivationPopupUI.cs

```diff
@@ -0,0 +1,86 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class ActivationPopupUI : MonoBehaviour
+{
+    [SerializeField]
+    private Animator anim;
+    [SerializeField]
+    private PlayerInput playerInput;
+
+    private void OnEnable()
+    {
+        if(playerInput != null)
+        {
+            playerInput.onControlsChanged += PlayerInput_onControlsChanged;
+
+            HandleInput(playerInput);
+        }
+    }
+
+    private void OnDisable()
+    {
+        if (playerInput != null)
+        {
+            playerInput.onControlsChanged -= PlayerInput_onControlsChanged;
+        }
+    }
+
+    private void Awake()
+    {
+        anim = GetComponent<Animator>();
+        playerInput = FindObjectOfType<PlayerInput>();
+    }
+
+    private void PlayerInput_onControlsChanged(PlayerInput input)
+    {
+        HandleInput(input);
+    }
+
+    public void HandleInput(PlayerInput input)
+    {
+        if (input.currentControlScheme == "Keyboard&Mouse") { ChangeUIPrompt("KEYBOARD"); }
+        if (input.currentControlScheme == "Gamepad")
+        {
+            if (Gamepad.current.device.name.Contains("XInput")) { ChangeUIPrompt("XINPUT"); }
+            else if (Gamepad.current.device.name.Contains("DualShock")) { ChangeUIPrompt("DUALSHOCK"); }
+            else if (Gamepad.current.device.name.Contains("Switch")) { ChangeUIPrompt("SWITCH"); }
+        }
+    }
+
+    public void ChangeUIPrompt(string deviceName)
+    {
+
+        switch (deviceName.ToUpper())
```

Comment:
ToUpper is an odd choice, but to each their own. Most people do ToLower instead

Reply:
changed to enum instead